src/main/host/descriptor/socket/inet/legacy_tcp.rs

```diff
@@ -151,7 +151,9 @@ impl LegacyTcpSocket {
             return None;
         }
 
-        Some(Packet::from_raw(packet))
+        let packet = Packet::from_raw(packet);
+        unsafe { c::packet_ref(packet.borrow_inner()) }
```

Comment:
Should `Packet::from_raw` make this call? Does it (and should it) `unref` on `Drop`?

Reply:
`Packet::from_raw` doesn't `ref`, but it does `unref` when dropped. The documentation for `from_raw` says that it takes ownership of the reference to the packet.